iothub/device/samples/PnpDeviceSamples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -100,12 +104,12 @@ public async Task PerformOperationsAsync(CancellationToken cancellationToken)
             // see if we have a device reported value for "serialNumber" and Thermostat2."initialValue".
             if (properties.Contains("serialNumber"))
             {
-                var serialNumberReported = properties["serialNumber"];
+                _logger.LogDebug($"Found property \"serialNumber\": {properties["serialNumber"]}");
             }
 
-            if (properties.Contains(Thermostat2) && (properties[Thermostat2] as Dictionary<string, object>).ContainsKey("initialValue"))
+            if (properties.Contains(Thermostat2) && ((JsonElement)properties[Thermostat2]).TryGetProperty("initialValue", out JsonElement initialStateThermostat2))
```

Comment:
right now the property collection will have <string, json-object-based-on-your-serializer>, which requires us to do this.
I don't like this. :(

Reply:
very difficult and unintuitive to retrieve the value of a component-level property

Reply:
Should we add PropertyCollection.Get<> methods? This would allow us to control the way the data is accessed and makes more sense with a component aware property

Get<RadObject>("propertyName", "componentName");
GetComponent<RadObject>("componentName");